crates/ra_hir_ty/src/method_resolution.rs

```diff
@@ -532,7 +532,10 @@ fn is_valid_candidate(
             let data = db.const_data(c);
             name.map_or(true, |name| data.name.as_ref() == Some(name)) && receiver_ty.is_none()
         }
-        _ => false,
+        AssocItemId::TypeAliasId(t) => {
```

Comment:
r? @flodiebold 

From the look of it, the old code looks correct, as it only uses the values namespace. I would expect two similar fuctions, one for consttants & methods, and one for types. The caller can then select the appropriate namespace

Reply:
Yeah, `method_resolution` is for values only, and associated types resolution has different rules. (You can't just write `S::Item` if `S` is a struct implementing `Iterator`, the only situation where you can write `T::AssociatedType` where `T` is a type is when `T` is a type parameter.)

Reply:
Might still be a good idea to replace `_` with explicit match & comment